CustomItems/Items/Scp5355.cs

```diff
@@ -0,0 +1,168 @@
+﻿
+namespace CustomItems.Items
+{
+    using System.Collections.Generic;
+    using System.ComponentModel;
+    using Exiled.API.Extensions;
+    using Exiled.API.Features;
+    using Exiled.CustomItems.API;
+    using Exiled.CustomItems.API.Features;
+    using Exiled.CustomItems.API.Spawn;
+    using Exiled.Events.EventArgs;
+    using Hints;
+    using MEC;
+    using UnityEngine;
+
+
+    public class Scp5355 : CustomWeapon
+    {
+        private readonly Dictionary<Player, Vector3> sizedPlayers = new Dictionary<Player, Vector3>();
+        private int mode = 1;
+
+        /// <inheritdoc/>
+        public override uint Id { get; set; } = 16;
+
+        /// <inheritdoc/>
+        public override string Name { get; set; } = "SCP-5355";
+
+        /// <inheritdoc/>
+        public override string Description { get; set; } = "SCP-5355 is a gun that can change the size of a person depending on the setting";
+
+        /// <inheritdoc/>
+        public override SpawnProperties SpawnProperties { get; set; } = new SpawnProperties
+        {
+            Limit = 1,
+            DynamicSpawnPoints = new List<DynamicSpawnPoint>
+            {
+                new DynamicSpawnPoint
+                {
+                    Chance = 100,
+                    Location = SpawnLocation.Inside173Armory,
+                },
+            },
+        };
+
+        /// <inheritdoc/>
+        public override Modifiers Modifiers { get; set; } = default;
+
+        /// <inheritdoc/>
+        public override float Damage { get; set; }
+
+        /// <inheritdoc/>
+        public override uint ClipSize { get; set; } = 10;
+
+        /// <summary>
+        /// Gets or sets the float modifier used to determine how much the player size is changed.
+        /// </summary>
+        [Description("The float modifier used to determine how much the player size is changed on each hit.")]
+        public float SizeChange { get; set; } = 0.1f;
+
+        /// <inheritdoc/>
+        protected override void SubscribeEvents()
+        {
+            Exiled.Events.Handlers.Player.Destroying += OnDestroying;
+            Exiled.Events.Handlers.Player.Died += OnDied;
+
+            base.SubscribeEvents();
+        }
+
+        /// <inheritdoc/>
+        protected override void UnsubscribeEvents()
+        {
+            Exiled.Events.Handlers.Player.Destroying -= OnDestroying;
+            Exiled.Events.Handlers.Player.Died -= OnDied;
+
+            base.UnsubscribeEvents();
+        }
+
+        /// <inheritdoc/>
+        protected override void OnReloading(ReloadingWeaponEventArgs ev)
```

Comment:
Keep in mind, that player without any ammo won't be able to change mode. You can try doing something about that.